### PR TITLE
Vreplication Cross-cell source: add support for cells in reshard

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**

### CODE REVIEW

File: go/test/endtoend/vreplication/cluster.go [hunk 1]
```diff
@@ -81,9 +84,20 @@ type Tablet struct {
...
 
+func init() {
+	originalVtdataroot = os.Getenv("VTDATAROOT")
+}
+
 func initGlobals() {
-	vtdataroot = os.Getenv("VTDATAROOT")
+	rand.Seed(time.Now().UTC().UnixNano())
+	dirSuffix := 100000 + rand.Intn(999999-100000) // 6 digits
+	vtdataroot = path.Join(originalVtdataroot, fmt.Sprintf("vreple2e_%d", dirSuffix))
```
Comment (deepthi) [inline go/test/endtoend/vreplication/cluster.go:94]: I see you are taking no chances with collisions :smile:

File: go/vt/vtctl/vtctl.go [hunk 2]
```diff
@@ -1864,12 +1866,12 @@ func commandReshard(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.F
...
 	source := strings.Split(subFlags.Arg(1), ",")
 	target := strings.Split(subFlags.Arg(2), ",")
-	return wr.Reshard(ctx, keyspace, workflow, source, target, *skipSchemaCopy)
+	return wr.Reshard(ctx, keyspace, workflow, source, target, *skipSchemaCopy, *cells, *tabletTypes)
 }
 
 func commandMoveTables(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.FlagSet, args []string) error {
 	workflow := subFlags.String("workflow", "", "Workflow name. Will be used to later migrate traffic.")
-	cell := subFlags.String("cell", "", "Cell to replicate from.")
+	cells := subFlags.String("cells", "", "Cell (comma-separated) to replicate from.")
```
Comment (deepthi) [inline go/vt/vtctl/vtctl.go:1874]:
```suggestion
	cells := subFlags.String("cells", "", "Cell(s) or CellsAlias(es) (comma-separated) to replicate from.")
```

File: go/vt/vtctl/vtctl.go [hunk 3]
```diff
@@ -310,10 +310,10 @@ var commands = []commandGroup{
 				"[-ping-tablets] <keyspace name>",
 				"Validates that all nodes reachable from the specified keyspace are consistent."},
 			{"Reshard", commandReshard,
-				"[-skip_schema_copy] <keyspace.workflow> <source_shards> <target_shards>",
+				"[-cells=<cells>] [-tablet_types=<source_tablet_types>] [-skip_schema_copy] <keyspace.workflow> <source_shards> <target_shards>",
 				"Start a Resharding process. Example: Reshard ks.workflow001 '0' '-80,80-'"},
```
Comment (deepthi) [inline go/vt/vtctl/vtctl.go:314]: Can you add `-cells` and `-tablet_types` to the example?

File: go/vt/wrangler/resharder.go [hunk 4]
```diff
@@ -47,6 +47,8 @@ type resharder struct {
 	targetMasters map[string]*topo.TabletInfo
 	vschema       *vschemapb.Keyspace
 	refStreams    map[string]*refStream
+	cell          string
```
Comment (deepthi) [inline go/vt/wrangler/resharder.go:50]: should document as "single cell or cellsAlias or comma-separated list of cells/cellsAliases"

File: go/vt/wrangler/resharder_test.go [hunk 5]
```diff
@@ -68,7 +68,7 @@ func TestResharderOneToMany(t *testing.T) {
 	env.tmc.expectVRQuery(200, "update _vt.vreplication set state='Running' where db_name='vt_ks'", &sqltypes.Result{})
 	env.tmc.expectVRQuery(210, "update _vt.vreplication set state='Running' where db_name='vt_ks'", &sqltypes.Result{})
 
-	err := env.wr.Reshard(context.Background(), env.keyspace, env.workflow, env.sources, env.targets, true)
+	err := env.wr.Reshard(context.Background(), env.keyspace, env.workflow, env.sources, env.targets, true, "", "")
```
Comment (deepthi) [inline go/vt/wrangler/resharder_test.go:71]: the unit test is not testing with any non-blank values. Should we not add cell / tablet_types to one of them?

File: go/test/endtoend/vreplication/vreplication_test.go [hunk 6]
```diff
@@ -131,16 +142,20 @@ func TestCellAliasVreplicationWorkflow(t *testing.T) {
...
 	verifyClusterHealth(t)
 	insertInitialData(t)
 	shardCustomer(t, true, []*Cell{cell1, cell2}, "alias")
-	// TODO: Test resharding once -cells option is added to Reshard command
-	//insertMoreCustomers(t, 16)
-	//reshardCustomer2to4Split(t)
-	//expectNumberOfStreams(t, vtgateConn, "Customer2to4", "sales", "product:0", 4)
+
+	insertMoreCustomers(t, 16)
+	reshardCustomer2to4Split(t, []*Cell{cell1, cell2}, "zone1")
```
Comment (deepthi) [inline go/test/endtoend/vreplication/vreplication_test.go:154]:
```suggestion
	reshardCustomer2to4Split(t, []*Cell{cell1, cell2}, "alias")
```

**Comment by deepthi** [comment]
@rohit-nayak-ps can you also enable these tests in CI? Then we can close #6453

File: go/test/endtoend/vreplication/cluster.go (hunk 1 above)
Comment (rohit-nayak-ps) [reply go/test/endtoend/vreplication/cluster.go:94]: paranoia with a small basis in reality ;-)

**Comment by rohit-nayak-ps** [comment]
> @rohit-nayak-ps can you also enable these tests in CI? Then we can close #6453

Github CI seems backed up today: tests have been queued up for a couple of hours. Will enable the repl e2e once tests pass. But I thought you had issues with the VM configuration in #6453? You mentioned breaking up the tests in the last Vitess call, but TestBasicVreplicationWorkflow was explicitly designed to run with the coded series of workflows to test different flow and edge cases.  

If I face the same issue, I will try tearing down unnecessary tablets in that test and see. Otherwise we may want to enable tests as a separate PR.

**Comment by deepthi** [comment]
According to the actions view, all tests completed and one failed. https://github.com/vitessio/vitess/actions?query=branch%3Arn-vrepl-crosscell-source
It is not reflecting correctly on the PR. I'm attempting to re-run the failed test.

**Comment by deepthi** [comment]
All tests passed including `unit_race` which had failed on first attempt. PR is not reflecting results from GH actions correctly.

**Pull request merged**